open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/vrp/retrieval/flow/VRPConsentRetrievalUtilTest.java

```diff
@@ -0,0 +1,432 @@
+
+        JSONArray cofConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+
+        Assert.assertNotNull(cofConsentData);
+    }
+
+    @Test
+    public void testGetConsentDataSetForVRPData() {
+
+        Mockito.doReturn(ConsentExtensionConstants.VRP).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.VRP_INITIATION).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+            JSONArray vrpConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+            Assert.assertNotNull(vrpConsentData);
+    }
+
+    @Test
+    public void testAppendDummyAccountID() {
+
+        JSONArray result = ConsentRetrievalUtil.appendDummyAccountID();
+
+        Assert.assertNotNull(result);
+
+        assertEquals(result.size(), 2);
+
+        JSONObject accountOne = (JSONObject) result.get(0);
+        assertEquals(accountOne.get("account_id"), "12345");
+        assertEquals(accountOne.get("display_name"), "Salary Saver Account");
+
+        JSONObject accountTwo = (JSONObject) result.get(1);
+        assertEquals(accountTwo.get("account_id"), "67890");
+        assertEquals(accountTwo.get("display_name"), "Max Bonus Account");
+    }
+
+    @Test(expectedExceptions = ConsentException.class)
+    public void testConsentDataWithInvalidJson() {
+        String invalidJsonString = "Invalid JSON String";
+        Mockito.when(consentResourceMock.getReceipt()).thenReturn(invalidJsonString);
+
+        try {
+            JSONArray consentDataJSON = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+            Assert.assertNotNull(consentDataJSON);
+        } catch (ConsentException e) {
+            // Parse the error message from the exception
+            JSONObject errorPayload = e.getPayload();
+            JSONArray errorsArray = (JSONArray) errorPayload.get("Errors");
+            JSONObject firstError = (JSONObject) errorsArray.get(0);
+            String errorMessage = (String) firstError.get("Message");
+
+            // Verify the error message
+            Assert.assertEquals(errorMessage, "Payload is not a JSON object");
+            throw e;
+        }
+    }
+
+    @Test(expectedExceptions = ConsentException.class)
+    public void testGetConsentDataSetForVRPDataParameter() {
+
+        Mockito.doReturn(ConsentExtensionConstants.VRP).when(consentResourceMock).getConsentType();
+        Mockito.doReturn(VRP_WITHOUT_DATA).when(consentResourceMock)
+                .getReceipt();
+        Mockito.doReturn(ConsentAuthorizeTestConstants.AWAITING_AUTH_STATUS).when(consentResourceMock)
+                .getCurrentStatus();
+
+        try {
+            JSONArray vrpConsentData = ConsentRetrievalUtil.getConsentData(consentResourceMock);
+            Assert.assertNotNull(vrpConsentData);
```

Comment:
ref : https://github.com/wso2/financial-open-banking/pull/38/files#r1528241432

Reply:
check below as well

Reply:
Fixed in [460b4c0]

Reply:
Fixed in[460b4c0]